neuropacks/nsds_nwb/nsds_nwb.py

```diff
@@ -47,54 +51,159 @@ def __init__(self, nwb_path):
+            Start time for loading data from an interval.
+        stop_time : float
+            Stop time for loading data from an interval.
 
-    def _load_poly(self):
-        """Load polytrode data, if available."""
-        poly = []
+        Returns
+        -------
+        A SimpleNamespace object.
+        '''
+        data_holder = []
         with NWBHDF5IO(self.nwb_path, 'r') as io:
             nwb = io.read()
             di = nwb.processing['preprocessing'].data_interfaces
             for n in di.keys():
-                if 'poly' in n.lower():
-                    poly.append(di[n].data[:])
-                    poly_rate = di[n].rate
-        if len(poly) == 1:
-            idxs = self.electrode_df['group_name'] == 'Poly'
+                if not n.startswith('wvlt_'):
+                    continue
+                if data_source in n.lower():
+                    rate = di[n].rate
+                    starting_time = di[n].starting_time
+                    if load_data:
+                        data_idx = slice_interval(start_time, stop_time,
+                                                  rate, t_offset=starting_time)
+                        data = di[n].data[data_idx]  # first axis is for the timepoints
+                        n_timepoints, n_channels, n_bands = data.shape
+                        if start_time is not None:
+                            starting_time = start_time
+                    else:
+                        data = None
+                        n_timepoints, n_channels, n_bands = di[n].data.shape
+                    data_holder.append(data)
+
+        name = data_source
+        if start_time is not None or stop_time is not None:
+            name = f'{name}_subset'
+
+        data_source_cased = NEURAL_DATA_SOURCES[data_source]
+        if len(data_holder) == 1:
+            idxs = self.electrode_df['group_name'] == data_source_cased
             good_electrodes = ~self.electrode_df['bad'].loc[idxs].values
-            self.poly = SimpleNamespace(data=poly[0],
-                                        rate=poly_rate,
-                                        good_electrodes=good_electrodes)
-        elif len(poly) == 0:
-            pass
+            return SimpleNamespace(name=name,
+                                   data=data_holder[0],
+                                   rate=rate,
+                                   n_timepoints=n_timepoints,
+                                   n_channels=n_channels,
+                                   n_bands=n_bands,
+                                   starting_time=starting_time,
+                                   good_electrodes=good_electrodes)
+        elif len(data_holder) == 0:
+            return None
         else:
-            raise ValueError('Multiple Poly sources found.')
+            raise ValueError(f'Multiple {data_source_cased} sources found.')
+
+    def _load_stimulus_waveform(self, start_time=None, stop_time=None):
```

Comment:
Get rid of this and just have `_get_stimulus_waveform()`?

Reply:
That works!